ipatests/test_integration/test_trust.py

```diff
@@ -992,3 +1052,242 @@ def test_server_option_with_unreachable_ad(self):
             self.remove_trust(self.ad)
+
+
+class TestNonPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with non posix AD trust
+    """
+    topology = 'line'
+    num_ad_domains = 1
+
+    @classmethod
+    def install(cls, mh):
+        if not cls.master.transport.file_exists('/usr/bin/rpcclient'):
+            raise pytest.skip("Package samba-client not available "
+                              "on {}".format(cls.master.hostname))
+        super(TestNonPosixAutoPrivateGroup, cls).install(mh)
+        cls.ad = cls.ads[0]
+        cls.ad_domain = cls.ad.domain.name
+        tasks.install_adtrust(cls.master)
+        tasks.sync_time(cls.master, cls.ad)
+        cls.existing_rangename = cls.ad_domain.upper() + "_id_range"
+
+    def test_add_nonposix_trust(self):
+        tasks.configure_dns_for_trust(self.master, self.ad)
+        tasks.establish_trust_with_ad(
+            self.master, self.ad_domain,
+            extra_args=['--range-type', 'ipa-ad-trust'])
+
+    def test_modify_auto_private_groups_value_default_trusted_range(self):
+        """
+        Modify existing range for default trusted AD domain range
+        with auto-private-groups set as true
+        """
+        domainusers = 'domain users'
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(self.existing_rangename, "true")
+        result = self.master.run_command(["id", nonposixuser])
+        uid_reg = (
+            r"^uid=(\d+)\({0}\)".format(nonposixuser)
+        )
+        gid_reg = (
+            r"gid=(\d+)\({0}\)".format(nonposixuser)
+        )
+        grp_reg = (
+            r"groups=(\d+)\({0}\),(\d+)\({1}@{2}\)".format
+            (nonposixuser, domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
+
+    def test_modify_auto_private_groups_value_trusted_range_hybrid(self):
+        """
+        Modify existing range for default trusted AD domain range
+        with auto-private-groups set as hybrid
+        """
+        domainusers = 'domain users'
+        nonposixuser = "nonposixuser@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group(self.existing_rangename, "hybrid")
+        result = self.master.run_command(["id", nonposixuser])
+        uid_reg = (
+            r"^uid=(\d+)\({0}\)".format(nonposixuser)
+        )
+        gid_reg = (
+            r"gid=(\d+)\({0}\)".format(nonposixuser)
+        )
+        grp_reg = (
+            r"groups=(\d+)\({0}\),(\d+)\({1}@{2}\)".format
+            (nonposixuser, domainusers, self.ad_domain)
+        )
+        assert re.search(uid_reg, result.stdout_text)
+        assert re.search(gid_reg, result.stdout_text)
+        assert re.search(grp_reg, result.stdout_text)
+
+    def test_idoverride_with_auto_private_group_false(self):
```

Comment:
This test is 99% identical to class TestPosixAutoPrivateGroup::test_idoverride_with_auto_private_group_false.
Please refactor (only the user differs between the 2 tests).